src/lib/storage/index/partial_hash/partial_hash_index.cpp

```diff
@@ -0,0 +1,82 @@
+#include "partial_hash_index.hpp"
+
+#include "storage/segment_iterate.hpp"
+
+namespace hyrise {
+
+PartialHashIndex::PartialHashIndex(const std::vector<std::pair<ChunkID, std::shared_ptr<Chunk>>>& chunks_to_index,
+                                   const ColumnID column_id)
+    : AbstractTableIndex{TableIndexType::PartialHash}, _column_id(column_id) {
+  Assert(!chunks_to_index.empty(), "PartialHashIndex requires chunks_to_index not to be empty.");
+  resolve_data_type(chunks_to_index.front().second->get_segment(_column_id)->data_type(),
+                    [&](const auto column_data_type) {
+                      using ColumnDataType = typename decltype(column_data_type)::type;
+                      _impl = std::make_shared<PartialHashIndexImpl<ColumnDataType>>(chunks_to_index, _column_id);
+                    });
+  insert_entries(chunks_to_index);
+}
+
+PartialHashIndex::PartialHashIndex(const DataType data_type, const ColumnID column_id)
```

Comment:
I don't understand what this constructor does. Where is it linked to a table? I just see a column ID.

Reply:
Right. We prevent the Table and PartialHashIndex from a circular dependency. We use a uni-directional dependency here: The table object stores a vector of table index pointers. To retrieve the created indexes of a given table, one has to call `get_table_indexes` of the table object. Thus, the index does not need to know which table it was created for.

This constructor is only called if a table index is to be created for a table without any chunks. In this case, the PartialHashIndex's constructor cannot retrieve the column's data type from a segment of the indexed column, so the DataType has to be provided explicitly.

I'll add a comment. Any feedback on that design?

Reply:
For what do we need empty indexes?

Reply:
If we need them, a comment summarizing your comment above would have helped.